src/Microsoft.ML.Data/Scorers/WhatTheFeatureScorer.cs

```diff
@@ -0,0 +1,457 @@
+﻿//------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Float = System.Single;
+
+[assembly: LoadableClass(typeof(IDataScorerTransform), typeof(WhatTheFeatureScorerTransform), typeof(WhatTheFeatureScorerTransform.Arguments),
+    typeof(SignatureDataScorer), "WhatTheFeature Scorer", "wtf", "WhatTheFeatureScorer", MetadataUtils.Const.ScoreColumnKind.WhatTheFeature)]
+
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(WhatTheFeatureScorerTransform), typeof(WhatTheFeatureScorerTransform.Arguments),
+    typeof(SignatureBindableMapper), "WhatTheFeature Mapper", "wtf", MetadataUtils.Const.ScoreColumnKind.WhatTheFeature)]
+
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(WhatTheFeatureScorerTransform), null, typeof(SignatureLoadModel),
+    "WhatTheFeature Mapper", WhatTheFeatureScorerTransform.MapperLoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// The What The Feature scorer is superset of a generic scorer.
+    /// It outputs score columns from Generic Scorer plus for given features provides vector of corresponding feature contributions.
+    /// </summary>
+    public sealed class WhatTheFeatureScorerTransform
```

Comment:
>WhatTheFeatureScorerTransform [](start = 24, length = 29)

This may have been considered funny at the time when this was an internal tool, but I feel we should come up with a different name for this externally.